### PR TITLE
csv : create CsvReader.go

### PR DESCRIPTION
**Pull request by florianjeanmart**

### CODE REVIEW

File: csv/CsvReader.go [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+type CsvReader struct {
+	File            string
+	NbRowToIgnore   int
+	RowWithHeader   int
+	ColumnsModel    []*CsvColumnDescription
+	data            [][]string
+	headerMap       map[string]int
+	structureErrors []*CsvAnalysisReportError
+	reportRows      map[int]*CsvAnalysisReportRow
```
[the remark this answers is not in the document]
Comment (florianjeanmart) [reply csv/CsvReader.go:18]: yes it's possible. I will do that

[the remark this answers is not in the document]
Comment (florianjeanmart) [reply csv/CsvReader.go:24]: Done. But by default, the separator is a comma (as the Csv acronym specify ;) )

File: csv/CsvReader.go [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+// NbRowToIgnore : the number of row to ignore at the beginning of the file : if the first line contains headers and the second line contains data, then the NbRowToIgnore should be 1
+// RowWithHeader : the number of the row with number : if this is the first line,then RowWithHeader shoud be 1
+// ColumnsModel : the structure of the columns
+// Separator : the data separator; optional, comma is used by default
+type CsvReader struct {
+	File            string
+	NbRowToIgnore   int
+	RowWithHeader   int
+	ColumnsModel    []*CsvColumnDescription
+	Separator       rune
```
Comment (Dbelataris) [inline csv/CsvReader.go:20]: @florianjeanmart would be easier to put a string here and make the conversion to rune in the `ReadCsv` func no ?

Comment (florianjeanmart) [reply csv/CsvReader.go:20]: I'm not sure : the type expected by the library is rune. If I use a string, I should convert the type with a risk of error. And as I tested, a rune is write as a byte : `','`, so nothing to complex.

Comment (Dbelataris) [reply csv/CsvReader.go:20]: @JensSkipr WDYT? a rune or a string ?

Comment (JensSkipr) [reply csv/CsvReader.go:20]: I agree on `rune`. Main reason is same as Florian mentioned, risk of error when converting the string to a rune.

**Review by Dbelataris — APPROVED** [review]

File: csv/CsvReader.go (hunk 2 above)
Comment (JensSkipr) [reply csv/CsvReader.go:20]: I assume they will never use it like this, but to give you an idea. This letter `é́́` is composed of 4 runes:
image

**Pull request merged**